### PR TITLE
feat(fixtures): fixtures data for items

### PR DESCRIPTION
**Pull request by MyuddinKhatri**
Issue - https://github.com/agritheory/test_utils/issues/10
Fixtures data for Items and Item Groups collected from beam and inventory_tools

### CODE REVIEW

**Comment by MyuddinKhatri** [comment]
@agritheory should we include item groups as well?

**Comment by agritheory** [comment]
@MyuddinKhatri Item Groups, yes. In the case of the "builder" functions, I think we can start including them in the repo too. I'm not certain what the function signature should be, but I think we may want to allow for the creation of specific items.

**Comment by MyuddinKhatri** [comment]
@agritheory I added builder functions to create Items and Item Groups. please check.

**Comment by MyuddinKhatri** [comment]
@agritheory I updated the fixtures to contain all the data in JSON itself so can be directly inserted. Please check this out.

**Review by Alchez — APPROVED** [review]

File: test_utils/utils/setup_fixtures.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+except Exception as e:
+	raise (e)
+
+def get_fixtures_data_from_file(filename):
+	app_dir = pathlib.Path(__file__).resolve().parent.parent / "fixtures"
+	if pathlib.Path.exists(app_dir / filename):
+		with open(app_dir / filename) as f:
+			return json.load(f)
+
+def create_items():
```
Comment (agritheory) [inline test_utils/utils/setup_fixtures.py:15]: @MyuddinKhatri Can you add the "shortlist" API we discussed?

File: test_utils/utils/setup_fixtures.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	items = get_fixtures_data_from_file(filename="items.json")
+	for item in items:
+		if frappe.db.exists("Item", item.get("item_code")):
+			continue
+		i = frappe.new_doc("Item")
+		i.update(item)
+		i.save()
+
+def create_item_groups():
```
Comment (agritheory) [inline test_utils/utils/setup_fixtures.py:24]: Shortlist API here too.

**Review by agritheory — CHANGES_REQUESTED** [review]

File: test_utils/utils/setup_fixtures.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+except Exception as e:
+	raise (e)
+
+def get_fixtures_data_from_file(filename):
+	app_dir = pathlib.Path(__file__).resolve().parent.parent / "fixtures"
+	if pathlib.Path.exists(app_dir / filename):
+		with open(app_dir / filename) as f:
+			return json.load(f)
+
+def create_items(settings, shortlist=None):
```
Comment (Alchez) [inline test_utils/utils/setup_fixtures.py:15]: @MyuddinKhatri Should the shortlist be connected to some parameter like company? Or is this to be implemented later?

Comment (agritheory) [reply test_utils/utils/setup_fixtures.py:15]: @Alchez Following an "explicit is better than implicit" strategy here, so shortlist will install the items with matching PKs. This API should be the same for all the install helpers.

Comment (MyuddinKhatri) [reply test_utils/utils/setup_fixtures.py:15]: @agritheory @Alchez to work with shortlist API should I modify function create_items() - we pass the list of item codes and function creates the items accordingly?

Comment (agritheory) [reply test_utils/utils/setup_fixtures.py:15]: @MyuddinKhatri Yes, that what I had in mind. I think the "shortlist" argument name is not great and now is the time to workshop it if we want to make it consistent for all of these functions. Maybe "only_create" is better? It's only a few more characters and is more explicit.

Comment (MyuddinKhatri) [reply test_utils/utils/setup_fixtures.py]: @agritheory @Alchez I updated the function now we can pass list of item_codes we want to create from the fixtures. please check.

**Review by Alchez — APPROVED** [review]

**Review by agritheory — APPROVED** [review]

**Pull request merged**